google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiManagementPanel.java

```diff
@@ -0,0 +1,278 @@
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.intellij.openapi.util.Couple;
+import com.intellij.ui.IdeBorderFactory;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.ui.table.JBTable;
+import com.intellij.util.ui.UIUtil;
+import java.awt.Color;
+import java.awt.Component;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import javax.swing.BorderFactory;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTable;
+import javax.swing.event.TableModelListener;
+import javax.swing.table.DefaultTableCellRenderer;
+import javax.swing.table.TableModel;
+
+/**
+ * Panel allowing management activity of cloud APIs including api enablement and service account
+ * creation.
+ */
+public class GoogleCloudApiManagementPanel {
+
+  private JPanel panel;
+  private ProjectSelector projectSelector;
+  private JLabel selectProjectLabel;
+  private JBScrollPane apisScrollPane;
+  private JPanel apisPanel;
+  private JPanel selectedApisPanel;
+  private JPanel selectedApisWrapper;
+  private JTable selectedApisTable;
+  private JPanel allApisPanel;
+  private JPanel allApisWrapper;
+  private JTable allApisTable;
+
+  private static final int MIN_CLOUD_API_COL_WIDTH = 300;
+  private static final int MIN_CLOUD_API_ENABLEMENT_COL_WIDTH = 100;
+  private static final int CLOUD_API_NAME_COL = 0;
+  private static final int CLOUD_API_ENABLEMENT_COL = 1;
+
+  GoogleCloudApiManagementPanel() {
+    apisPanel.setBorder(BorderFactory.createEmptyBorder());
+
+    selectedApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    selectedApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.selectedapis.title")));
+
+    allApisWrapper.setBorder(IdeBorderFactory.createBorder());
+    allApisPanel.setBorder(
+        IdeBorderFactory.createTitledBorder(
+            GctBundle.message("cloud.apis.management.allapis.title")));
+
+    if (projectSelector.getSelectedProject() != null) {
+      showApiEnablementUi();
+    }
+    projectSelector.addProjectSelectionListener(project -> showApiEnablementUi());
+  }
+
+  private void showApiEnablementUi() {
+    apisScrollPane.setVisible(true);
+    selectProjectLabel.setVisible(false);
+  }
+
+  /** Initializes the table models of the API tables based on the user's library selection. */
+  // TODO(eshaul) its very important that we add an equals method to CloudLibrary so that the set
+  // operation is reliable
+  void init(List<CloudLibrary> allLibraries, Set<CloudLibrary> selectedLibraries) {
+    selectedApisTable.setModel(new ApisTableModel(selectedLibraries));
+    selectedApisTable
```

Comment:
It feels a bit off that we mix the code for updating the model of the tables with data and then resize the column.
How about we initialize the table with empty model in `createUIComponents` and apply column sizing/formatting in there as well, and use `init` for data operations only?

Reply:
sgtm